hs_core/models.py

```diff
@@ -1510,6 +1531,30 @@ def delete_all_elements(self):
         self.sources.all().delete()
         self.relations.all().delete()
 
+    def copy_all_elements_to(self, tgt_res):
+        from hydroshare.utils import current_site_url
+        from django.forms.models import model_to_dict
```

Comment:
Will move the second import to top level, but the first import cannot be moved to top level due to circular import, tried to overcome circular import, but could not succeed at this point. At some point in time, we may have to create a branch just to resolve this circular import issue since it is not possible to import methods from hydroshare.utils, hydroshare.resource, etc., in hs_core/models.py since all those modules imported models.py.